dataproc/src/main/java/SubmitJob.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START dataproc_submit_job]
+import com.google.cloud.dataproc.v1.Job;
+import com.google.cloud.dataproc.v1.JobControllerClient;
+import com.google.cloud.dataproc.v1.JobControllerSettings;
+import com.google.cloud.dataproc.v1.JobPlacement;
+import com.google.cloud.dataproc.v1.JobStatus;
+import com.google.cloud.dataproc.v1.PySparkJob;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public class SubmitJob {
+
+  public static void submitJob() throws IOException, InterruptedException, ExecutionException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String region = "your-project-region";
+    String clusterName = "your-cluster-name";
+    String jobFilePath = "your-job-file-path";
+    submitJob(projectId, region, clusterName, jobFilePath);
+  }
+
+  public static void submitJob(
+      String projectId, String region, String clusterName, String jobFilePath) throws IOException {
+    String myEndpoint = String.format("%s-dataproc.googleapis.com:443", region);
+
+    // Configure the settings for the cluster controller client
+    JobControllerSettings jobControllerSettings =
+        JobControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+    // Create a job controller client with the configured settings. The client only needs to be
+    // created once and can be reused for multiple requests. Using a try-with-resources
+    // closes the client, but this can also be done manually with the .close() method.
+    try (JobControllerClient jobControllerClient =
+        JobControllerClient.create(jobControllerSettings)) {
+
+      // Configure the settings for our job
+      JobPlacement jobPlacement = JobPlacement.newBuilder().setClusterName(clusterName).build();
+      PySparkJob pySparkJob = PySparkJob.newBuilder().setMainPythonFileUri(jobFilePath).build();
+      Job job = Job.newBuilder().setPlacement(jobPlacement).setPysparkJob(pySparkJob).build();
+
+      Job jobOperation = jobControllerClient.submitJob(projectId, region, job);
+
+      String jobId = jobOperation.getReference().getJobId();
+
+      while (true) {
```

Comment:
Confused by this - it's a `while(true)` with either if conditions leading to a break, meaning it never iterates. Just remove the loop?

Reply:
`.submitJob()` is async and doesn't return a future object. `Job.getJob` makes an api request to get the current status of the job from the cluster, so the `while` loop blocks the program until the job ends via completion or an error. To make this clearer, I can reword this comment and include it in the sample and/or add a `Thread.sleep()` to query every 5-10 seconds rather than continuously. What do you think?

Reply:
Ah okay, that make more sense. It's unfortunate this isn't using the LongRunningOperation pattern of similar endpoints. 

We definitely need to add a comment and a sleep (of at least 1s) between API polls. 

Additionally, this while loop as a chance to hold indefinitely because there are other terminal states that aren't being checked here. Checking from the [`JobStatus.State`](https://googleapis.dev/java/google-api-grpc/latest/com/google/cloud/dataproc/v1/JobStatus.State.html) docs, it looks like `ATTEMPT_FAILURE`, `CANCELLED`, `DONE`, `ERROR`, and possibly `STATE_UNSPECIFIED` are all terminal. There probably needs to be a timeout function as well (especially since there are a couple of states that might not be terminal).

How would you feel about trying something like this (guessing on the correct timeunits)?
```java
CompleteableFuture<Job> finishedJobFuture = CompletableFuture.supplyAsync(() -> {
  // States that indicate a job is "finished"
  Set terminalStates = Set.of(JobStatus.State.ATTEMPT_FAILURE, JobStatus.State.CANCELLED, JobStatus.State.DONE, JobStatus.State.ERROR)
  while(true){
    // Poll the service periodically until the Job is in a finished state.
    Job job = jobControllerClient.getJob(projectId, region, jobId);
    if(terminalStates.contains(job.getStatus().getState()) {
      return job
    }
    // Wait a second in between polling attempts.
   TimeUnit.SECONDS.sleep(1);
  }
})

// Wait up to 10 minutes for the job to be finished.
Job finishedJob = finishedJobFuture.get(10, TimeUnit.Minutes);

// Verify the job was successful.
if (!jobInfo.getStatus().getState().equals(JobStatus.State.DONE)) {
          System.out.printf("Job %s did not finish successfully: %s", jobId, jobInfo.getStatus().getDetails());
}
System.out.printf("Job %s finished successfully.", jobId);
```